### PR TITLE
Add core externally used apis to ref assembly

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/linker/Linker/OverrideInformation.cs [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
 		public OverrideInformation (MethodDefinition @base, MethodDefinition @override, InterfaceImplementation matchingInterfaceImplementation = null)
 		{
 			Base = @base;
 			Override = @override;
 			MatchingInterfaceImplementation = matchingInterfaceImplementation;
 		}
 
+		public MethodDefinition Base { get; private set; }
+		public MethodDefinition Override { get; private set; }
+		public InterfaceImplementation MatchingInterfaceImplementation { get; private set; }
```
Comment (sbomer) [inline src/linker/Linker/OverrideInformation.cs]: Why add setters? It looks like we can keep these readonly since they're only set in the ctor.

Comment (marek-safar) [reply src/linker/Linker/OverrideInformation.cs]: It's more resilient, one can add or change set/get behaviour if needed without breaking the contract.

Comment (sbomer) [reply src/linker/Linker/OverrideInformation.cs]: Adding getters makes sense to me - I am specifically asking about the setters. If you only add a getter, the underlying field can still be readonly without changing the contract, and you can always introduce setters later if needed: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyDMABGgQMIEDeOB1RhARgPYMQEBCFBA5gKbADcBAL5Ua+UgAoAlBRE0a7ALwFgAJwCu3PrKE5BQA=

Comment (marek-safar) [reply src/linker/Linker/OverrideInformation.cs]: I see what you mean

File: src/linker/ref/Linker/Annotations.cs [hunk 2]
```diff
@@ -2,17 +2,27 @@
...
 using Mono.Cecil;
-using Mono.Cecil.Cil;
 
 namespace Mono.Linker {
 
 	public partial class AnnotationStore
 	{
 		internal AnnotationStore() {}
+
+		public List<OverrideInformation> GetOverrides (MethodDefinition method) { throw null; }
```
Comment (vitek-karas) [inline src/linker/ref/Linker/Annotations.cs]: Should we try to change these APIs to be a bit more restrictive - given that they are becoming an official public API.
For example this one could return `IEnumerable`. And the `GetCustomAnnotations` could be replaced by something like `TryGetCustomAnnotation(object key, IMetadataTokenProvider item, out object value)` and `SetCustomAnnotation(object key, IMetadataTokenProvider item, object value)`. Basically trying to hide the implementation details.

File: src/linker/ref/Linker/OverrideInformation.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class OverrideInformation
+	{
+		internal OverrideInformation ()
```
Comment (vitek-karas) [inline src/linker/ref/Linker/OverrideInformation.cs]: Why do we need to declare an internal .ctor in a ref assembly?

Comment (marek-safar) [reply src/linker/ref/Linker/OverrideInformation.cs]: There were errors about missing ctor in apidiff tool

File: src/linker/ref/Linker/Annotations.cs (hunk 2 above)
Comment (marek-safar) [reply src/linker/ref/Linker/Annotations.cs]: That makes sense to me (will change that in follow up PR)

File: src/linker/ref/Linker/OverrideInformation.cs (hunk 3 above)
Comment (sbomer) [reply src/linker/ref/Linker/OverrideInformation.cs]: (Specifically if you don't declare a ctor, the ref assembly will have the implicit public ctor, which doesn't exist in the implementation.)

Comment (vitek-karas) [reply src/linker/ref/Linker/OverrideInformation.cs]: Good to know - thanks!

**Pull request merged**